### ISSUE TITLE
On MongoDB Validation Error for a different issue Decimals incorrectly fail 'mongo-schemer' validation

### ISSUE THREAD

**Issue by matthew1534** [devetry/mongo-schemer#9]
In the below example the first error that `mongo-schemer` is returning is correct and is the only error causing the MongoDB validation. The second decimal error is incorrect.

```
[ 
  { 
    keyword: 'bsonType',
    params: { bsonType: 'int' },
    message: 'should be int got asdsa',
    dataPath: '.tiers[0].qty',
    schemaPath: '#/properties/tiers/items/properties/qty/bsonType' 
  },
  { 
    keyword: 'bsonType',
    params: { bsonType: 'decimal' },
    message: 'should be decimal got 1.2',
    dataPath: '.tiers[0].price',
    schemaPath: '#/properties/tiers/items/properties/price/bsonType' 
  }
]
```

The decimal value is being inserted using the `Decimal128.fromString()` function from https://www.npmjs.com/package/mongodb . Here is the example insertOne that caused this issue:
```
await db.collection(`offers`).insertOne({ 
  tiers: [
    {
      qty: "asdsa", // purposely wrong - should be an integer
      price: Decimal128.fromString(`1.2`) // correct!
    }
  ]
});
```

This is the `$jsonSchema` for these fields:
```
tiers: {
  bsonType: [`array`],
  items: {
    bsonType: [`object`],
    additionalProperties: false,
    properties: {
      qty: {
        bsonType: `int`
      },
      price: {
        bsonType: `decimal`,
        minimum: 0
      }
    } 
  }
}
```

Potentially an AJV issue rather than 'mongo-schemer' issue. Would using somthing like https://github.com/epoberezkin/ajv/blob/master/CUSTOM.md help to get around this issue?

P.S. love this module - saves so much time - thanks. Will have a go at fixing this issue until I hear back!

**Comment by bgschiller** [comment]
Hi @matthew1534  Thanks for the kind words! I'm happy to review and merge a PR for this issue.

It looks like a fix could be applied either here or in the [ajv-bsontype](https://github.com/BoLaMN/ajv-bsontype) library that we use.

**Comment by bgschiller** [comment]
oh! and yes, I suspect that custom keyword link is exactly what we need to solve the problem.

**Comment by matthew1534** [comment]
Great stuff - I'm busy for the next 2-3 weeks, once I get the time I'll try to fix and send across a PR!

**Comment by matthew1534** [comment]
@bgschiller FYI I've created a PR for this in [ajv-bsontype](https://github.com/BoLaMN/ajv-bsontype): https://github.com/BoLaMN/ajv-bsontype/pull/4

**Comment by bgschiller** [comment]
Great! We'll upgrade to the new version once it's released.

**Comment by BoLaMN** [comment]
fix released in 1.0.7
